tests/test_clean.py

```diff
@@ -452,3 +688,27 @@ def test_basics(self):
             cleaner.clean('a <br/><span style="color:red">test</span>') ==
             'a <br><span style="">test</span>'
         )
+
+    def test_filters(self):
+        # Create a Filter that changes all the attr values to "moo"
+        class MooFilter(Filter):
+            def __iter__(self):
+                for token in Filter.__iter__(self):
+                    if token['type'] in ['StartTag', 'EmptyTag'] and token['data']:
+                        for attr, value in token['data'].items():
+                            token['data'][attr] = 'moo'
+
+                    yield token
+
+        ATTRS = {
+            'img': ['rel', 'src']
+        }
+        TAGS = ['img']
+
+        cleaner = Cleaner(tags=TAGS, attributes=ATTRS, filters=[MooFilter])
+
+        dirty = 'this is cute! <img src="http://example.com/puppy.jpg" rel="nofollow">'
+        assert (
+            cleaner.clean(dirty) ==
+            'this is cute! <img rel="moo" src="moo">'
+        )
```

Comment:
This was a separate function, but it uses `Cleaner`, so I figured I'd move it to the `TestCleaner` test class so all the `Cleaner` stuff is one place.